### PR TITLE
[SC-623] Rename all instances of nst to usds

### PR DESCRIPTION
**Pull request by hexonaut**

### CODE REVIEW

File: test/fork/MigrationActions.t.sol [hunk 1]
```diff
@@ -79,132 +79,132 @@ contract MigrationActionsIntegrationTestBase is Test {
...
             sdai.totalAssets() +
-            snst.totalAssets() -
-            nst.balanceOf(SNST);
+            susds.totalAssets() -
+            usds.balanceOf(SUSDS);
     }
 
 }
 
 contract MigrateDaiToNstIntegrationTest is MigrationActionsIntegrationTestBase {
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `MigrateDaiToUsdsIntegrationTest`

File: test/fork/MigrationActions.t.sol [hunk 2]
```diff
@@ -79,132 +79,132 @@ contract MigrationActionsIntegrationTestBase is Test {
...
 
         deal(DAI, user, amount);  // Use `deal` to get a higher DAI amount
 
-        _runMigrateDAIToNSTTest(amount);
+        _runMigrateDAIToUSDSTest(amount);
     }
 
 }
 
 contract MigrateDaiToSNstIntegrationTest is MigrationActionsIntegrationTestBase {
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `MigrateDaiToSUsdsIntegrationTest`

File: test/fork/MigrationActions.t.sol [hunk 3]
```diff
@@ -79,132 +79,132 @@ contract MigrationActionsIntegrationTestBase is Test {
...
 
         deal(DAI, user, amount);
 
-        _runMigrateDAIToSNSTTest(amount);
+        _runMigrateDAIToSUSDSTest(amount);
     }
 
 }
 
 contract MigrateSDaiAssetsToNstIntegrationTest is MigrationActionsIntegrationTestBase {
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `MigrateSDaiAssetsToUsdsIntegrationTest`

File: test/fork/MigrationActions.t.sol [hunk 4]
```diff
@@ -286,64 +286,64 @@ contract MigrateSDaiSharesToNstIntegrationTest is MigrationActionsIntegrationTes
...
 
         deal(DAI, user, amount);
 
-        _runMigrateSDAISharesToNSTTest(amount);
+        _runMigrateSDAISharesToUSDSTest(amount);
     }
 
 }
 
 contract MigrateSDaiAssetsToSNstIntegrationTest is MigrationActionsIntegrationTestBase {
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `MigrateSDaiAssetsToSUsdsIntegrationTest`

File: test/fork/MigrationActions.t.sol [hunk 5]
```diff
@@ -354,66 +354,66 @@ contract MigrateSDaiAssetsToSNstIntegrationTest is MigrationActionsIntegrationTe
         uint256 debt           = vat.debt();
         uint256 sumSupply      = _getSumSupply();
 
-        actions.migrateSDAIAssetsToSNST(user, amount);
+        actions.migrateSDAIAssetsToSUSDS(user, amount);
 
         uint256 newUserSDaiAssets = sdai.convertToAssets(sdai.balanceOf(user));
-        uint256 userSNstAssets    = snst.convertToAssets(snst.balanceOf(user));
-        uint256 expectedDebt      = debt + daiDripAmount + nstDripAmount * 1e27;
+        uint256 userSNstAssets    = susds.convertToAssets(susds.balanceOf(user));
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `userSUsdsAssets`

File: test/fork/MigrationActions.t.sol [hunk 6]
```diff
@@ -354,66 +354,66 @@ contract MigrateSDaiAssetsToSNstIntegrationTest is MigrationActionsIntegrationTe
...
+        uint256 expectedDebt      = debt + daiDripAmount + usdsDripAmount * 1e27;
 
-        assertApproxEqAbs(userSNstAssets,    amount,                  2);  // User gets specified amount of sNST (conversion rounding x2)
+        assertApproxEqAbs(userSNstAssets,    amount,                  2);  // User gets specified amount of sUSDS (conversion rounding x2)
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `userSUsdsAssets`

File: test/fork/MigrationActions.t.sol [hunk 7]
```diff
@@ -354,66 +354,66 @@ contract MigrateSDaiAssetsToSNstIntegrationTest is MigrationActionsIntegrationTe
...
 
         deal(DAI, user, amount);
 
-        _runMigrateSDAIAssetsToSNSTTest(amount);
+        _runMigrateSDAIAssetsToSUSDSTest(amount);
     }
 
 }
 
 contract MigrateSDaiSharesToSNstIntegrationTest is MigrationActionsIntegrationTestBase {
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `MigrateSDaiSharesToSUsdsIntegrationTest`

File: test/fork/MigrationActions.t.sol [hunk 8]
```diff
@@ -425,98 +425,98 @@ contract MigrateSDaiSharesToSNstIntegrationTest is MigrationActionsIntegrationTe
         uint256 debt      = vat.debt();
         uint256 sumSupply = _getSumSupply();
 
-        actions.migrateSDAISharesToSNST(user, sdai.balanceOf(user));
+        actions.migrateSDAISharesToSUSDS(user, sdai.balanceOf(user));
 
         uint256 newUserSDaiAssets = sdai.convertToAssets(sdai.balanceOf(user));
-        uint256 userSNstAssets    = snst.convertToAssets(snst.balanceOf(user));
-        uint256 expectedDebt      = debt + daiDripAmount + nstDripAmount * 1e27;
+        uint256 userSNstAssets    = susds.convertToAssets(susds.balanceOf(user));
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `userSUsdsAssets`

File: test/fork/MigrationActions.t.sol [hunk 9]
```diff
@@ -425,98 +425,98 @@ contract MigrateSDaiSharesToSNstIntegrationTest is MigrationActionsIntegrationTe
...
+        uint256 expectedDebt      = debt + daiDripAmount + usdsDripAmount * 1e27;
 
-        assertApproxEqAbs(userSNstAssets,    userSDaiAssets, 2);  // User gets specified amount of sNST (conversion rounding x1)
+        assertApproxEqAbs(userSNstAssets,    userSDaiAssets, 2);  // User gets specified amount of sUSDS (conversion rounding x1)
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `userSUsdsAssets`

File: test/fork/MigrationActions.t.sol [hunk 10]
```diff
@@ -425,98 +425,98 @@ contract MigrateSDaiSharesToSNstIntegrationTest is MigrationActionsIntegrationTe
...
+        _runMigrateSDAISharesToSUSDSTest(amount);
     }
 
 }
 
 
-contract DowngradeNSTToDAIIntegrationTest is MigrationActionsIntegrationTestBase {
+contract DowngradeUSDSToDAIIntegrationTest is MigrationActionsIntegrationTestBase {
 
     function _getNst(address receiver, uint256 amount) internal {
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `_getUsds`

File: test/fork/MigrationActions.t.sol [hunk 11]
```diff
@@ -425,98 +425,98 @@ contract MigrateSDaiSharesToSNstIntegrationTest is MigrationActionsIntegrationTe
...
-        assertEq(dai.balanceOf(user), amount);
+        assertEq(usds.balanceOf(user), 0);
+        assertEq(dai.balanceOf(user),  amount);
     }
 
-    function test_downgradeNSTToDAI() public assertDebtStateDoesNotChange {
+    function test_downgradeUSDSToDAI() public assertDebtStateDoesNotChange {
         uint256 amount = 1000 ether;
 
         _getNst(user, amount);
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `_getUsds`

File: test/fork/MigrationActions.t.sol [hunk 12]
```diff
@@ -425,98 +425,98 @@ contract MigrateSDaiSharesToSNstIntegrationTest is MigrationActionsIntegrationTe
...
 
-        _runDowngradeNSTToDAITest(amount);
+        _runDowngradeUSDSToDAITest(amount);
     }
 
-    function testFuzz_downgradeNSTToDAI(uint256 amount) public assertDebtStateDoesNotChange {
+    function testFuzz_downgradeUSDSToDAI(uint256 amount) public assertDebtStateDoesNotChange {
         amount = _bound(amount, 0, dai.balanceOf(DAI_WHALE));
 
         _getNst(user, amount);
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: `_getUsds`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Some more updates needed

File: src/MigrationActions.sol [hunk 13]
```diff
@@ -15,128 +15,128 @@ interface VatLike {
...
+        usdsJoin = JoinLike(_usdsJoin);
+        vat      = daiJoin.vat();
 
         // Infinite approvals
-        dai.approve(_daiJoin, type(uint256).max);
-        nst.approve(_nstJoin, type(uint256).max);
-        nst.approve(_snst,    type(uint256).max);
+        dai.approve(_daiJoin,  type(uint256).max);
+        usds.approve(_usdsJoin, type(uint256).max);
+        usds.approve(_susds,    type(uint256).max);
```
Comment (lucas-manuel) [inline src/MigrationActions.sol]: Alignment

File: src/MigrationActions.sol [hunk 14]
```diff
@@ -15,128 +15,128 @@ interface VatLike {
...
      * @param  receiver The receiver of `dai`.
-     * @param  assetsIn The amount of `nst` to downgrade.
+     * @param  assetsIn The amount of `usds` to downgrade.
      */
-    function downgradeNSTToDAI(address receiver, uint256 assetsIn) external {
-        nst.transferFrom(msg.sender, address(this), assetsIn);
-        nstJoin.join(address(this), assetsIn);
+    function downgradeUSDSToDAI(address receiver, uint256 assetsIn) external {
+        usds.transferFrom(msg.sender, address(this), assetsIn);
+        usdsJoin.join(address(this), assetsIn);
```
Comment (lucas-manuel) [inline src/MigrationActions.sol:129]: Alignment

File: src/MigrationActions.sol [hunk 15]
```diff
@@ -15,128 +15,128 @@ interface VatLike {
...
 
     /**********************************************************************************************/
     /*** Internal helper functions                                                              ***/
     /**********************************************************************************************/
 
-    function _migrateDAIToNST(address receiver, uint256 amount) internal {
+    function _migrateDAIToUSDS(address receiver, uint256 amount) internal {
         daiJoin.join(address(this), amount);
-        nstJoin.exit(receiver,      amount);
+        usdsJoin.exit(receiver,      amount);
```
Comment (lucas-manuel) [inline src/MigrationActions.sol]: Alignment

File: test/fork/MigrationActions.t.sol [hunk 16]
```diff
@@ -286,64 +286,64 @@ contract MigrateSDaiSharesToNstIntegrationTest is MigrationActionsIntegrationTes
...
-        // Warp to accrue value in both sDAI and sNST after drip is called on sDAI deposit
+        // Warp to accrue value in both sDAI and sUSDS after drip is called on sDAI deposit
         skip(2 hours);
 
         // Get the expected amount to be sucked from the vat on `drip` in withdraw
-        // and deposit calls in sDAI and sNST respectively
+        // and deposit calls in sDAI and sUSDS respectively
         uint256 snapshot      = vm.snapshot();
-        uint256 nstBalance    = nst.balanceOf(SNST);
+        uint256 usdsBalance    = usds.balanceOf(SUSDS);
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol]: Alignment

File: test/fork/MigrationActions.t.sol [hunk 17]
```diff
@@ -286,64 +286,64 @@ contract MigrateSDaiSharesToNstIntegrationTest is MigrationActionsIntegrationTes
...
         uint256 preDripPotDai = vat.dai(POT);
-        snst.drip();
+        susds.drip();
         pot.drip();
-        uint256 nstDripAmount = nst.balanceOf(SNST) - nstBalance;
+        uint256 usdsDripAmount = usds.balanceOf(SUSDS) - usdsBalance;
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol:346]: Alignment

File: test/fork/MigrationActions.t.sol (hunk 5 above)
Comment (lucas-manuel) [reply test/fork/MigrationActions.t.sol]: Alignment

File: test/fork/MigrationActions.t.sol (hunk 6 above)
Comment (lucas-manuel) [reply test/fork/MigrationActions.t.sol]: Alignment

File: test/fork/MigrationActions.t.sol [hunk 18]
```diff
@@ -354,66 +354,66 @@ contract MigrateSDaiAssetsToSNstIntegrationTest is MigrationActionsIntegrationTe
...
+        // and deposit calls in sDAI and sUSDS respectively
         uint256 snapshot      = vm.snapshot();
-        uint256 nstBalance    = nst.balanceOf(SNST);
+        uint256 usdsBalance   = usds.balanceOf(SUSDS);
         uint256 preDripPotDai = vat.dai(POT);
-        snst.drip();
+        susds.drip();
         pot.drip();
-        uint256 nstDripAmount = nst.balanceOf(SNST) - nstBalance;
+        uint256 usdsDripAmount = usds.balanceOf(SUSDS) - usdsBalance;
```
Comment (lucas-manuel) [inline test/fork/MigrationActions.t.sol:416]: Alignment

File: test/fork/MigrationActions.t.sol (hunk 9 above)
Comment (lucas-manuel) [reply test/fork/MigrationActions.t.sol]: Alignment

File: test/fork/MigrationActions.t.sol (hunk 8 above)
Comment (lucas-manuel) [reply test/fork/MigrationActions.t.sol]: Alignment

File: test/unit/MigrationActions.t.sol [hunk 19]
```diff
@@ -14,55 +14,55 @@ import { MigrationActions } from "src/MigrationActions.sol";
...
 
     address receiver = makeAddr("receiver");
 
     function setUp() public {
-        dai  = new MockERC20("DAI", "DAI", 18);
-        nst  = new MockERC20("NST", "NST", 18);
-        sdai = new ERC4626Mock(dai, "sDAI", "sDAI", 18);
-        snst = new ERC4626Mock(nst, "sNST", "sNST", 18);
+        dai   = new MockERC20("DAI", "DAI", 18);
+        usds  = new MockERC20("USDS", "USDS", 18);
```
Comment (lucas-manuel) [inline test/unit/MigrationActions.t.sol]: Align params

File: test/unit/MigrationActions.t.sol [hunk 20]
```diff
@@ -14,55 +14,55 @@ import { MigrationActions } from "src/MigrationActions.sol";
...
+        sdai  = new ERC4626Mock(dai, "sDAI", "sDAI", 18);
+        susds = new ERC4626Mock(usds, "sUSDS", "sUSDS", 18);
```
Comment (lucas-manuel) [inline test/unit/MigrationActions.t.sol]: Align params

File: README.md [hunk 21]
```diff
@@ -143,9 +143,9 @@ actions.redeemAndSwap(address(this), bal, sDAI.convertToAssets(bal));
 
 ## MigrationActions
 
-This contract is used to upgrade from DAI, sDAI to NST, sNST. Also contains a downgrade path for NST -> DAI for backwards compatibility.
+This contract is used to upgrade from DAI, sDAI to USDS, sUSDS. Also contains a downgrade path for USDS -> DAI for backwards compatibility.
 
-Below is a diagram that outlines the migration path from sDAI to sNST. This migration path is the most complex and all other paths are subsets of this one.
+Below is a diagram that outlines the migration path from sDAI to sUSDS. This migration path is the most complex and all other paths are subsets of this one.
 
 ![sDai to sNst](https://github.com/user-attachments/assets/a7b74251-f7bb-46e1-ac77-88a1e1c452b5)
```
Comment (lucas-manuel) [inline README.md]: `sUsds`

Comment (lucas-manuel) [reply README.md]: Need to update diagram naming

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: src/MigrationActions.sol (hunk 13 above)
Comment (hexonaut) [reply src/MigrationActions.sol]: fixed

File: src/MigrationActions.sol (hunk 15 above)
Comment (hexonaut) [reply src/MigrationActions.sol]: fixed

File: test/fork/MigrationActions.t.sol (hunk 16 above)
Comment (hexonaut) [reply test/fork/MigrationActions.t.sol]: fixed

File: test/fork/MigrationActions.t.sol (hunk 17 above)
Comment (hexonaut) [reply test/fork/MigrationActions.t.sol:346]: fixed

File: test/fork/MigrationActions.t.sol (hunk 18 above)
Comment (hexonaut) [reply test/fork/MigrationActions.t.sol:416]: fixed

File: test/unit/MigrationActions.t.sol (hunk 19 above)
Comment (hexonaut) [reply test/unit/MigrationActions.t.sol:31]: were these aligned before? It's a bit awkward

Comment (hexonaut) [reply test/unit/MigrationActions.t.sol]: fixed

File: test/unit/MigrationActions.t.sol (hunk 20 above)
Comment (hexonaut) [reply test/unit/MigrationActions.t.sol]: fixed

File: src/MigrationActions.sol (hunk 15 above)
Comment (lucas-manuel) [reply src/MigrationActions.sol]: Still needs to be fixed

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Review by barrutko — APPROVED** [review]

**Pull request merged**